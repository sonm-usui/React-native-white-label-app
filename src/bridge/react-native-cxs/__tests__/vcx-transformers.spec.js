// @flow

import {
  convertAgencyConfigToVcxProvision,
  convertVcxProvisionResultToUserOneTimeInfo,
  convertCxsInitToVcxInit,
  convertInvitationToVcxConnectionCreate,
  convertLegacyClaimOfferToAppClaimOffer,
} from '../vcx-transformers'
import {
  vcxProvisionResult,
  agencyUrl,
  agencyDID,
  agencyVerificationKey,
  poolConfig,
  paymentMethod,
  userOneTimeInfo,
  getTestInvitationPayload,
  vcxClaimOffer,
} from '../../../../__mocks__/static-data'

const agencyPoolConfig = {
  agencyUrl,
  agencyDID,
  agencyVerificationKey,
  poolConfig,
  paymentMethod,
}

const initWithGenesisPathConfig = {
  ...userOneTimeInfo,
  ...agencyPoolConfig,
  genesis_path: 'genesis_path',
}

const walletPoolName = {
  walletName: 'walletName',
  poolName: 'poolName',
}

jest.mock('react-native-device-info', () => {
  return {
    getDeviceName: () => 'test-name',
    getModel: () => 'test-model',
  }
})

describe('transformer:VCX', () => {
  it('convertAgencyConfigToVcxProvision', async () => {
    const vcxProvisionConfig = await convertAgencyConfigToVcxProvision(
      agencyPoolConfig,
      walletPoolName
    )
    expect(vcxProvisionConfig).toMatchSnapshot()
  })

  it('convertVcxProvisionResultToUserOneTimeInfo', () => {
    expect(
      convertVcxProvisionResultToUserOneTimeInfo(vcxProvisionResult)
    ).toMatchSnapshot()
  })

  it('convertCxsInitToVcxInit', async () => {
    const vcxInitConfig = await convertCxsInitToVcxInit(
      initWithGenesisPathConfig,
      walletPoolName
    )
    expect(vcxInitConfig).toMatchSnapshot()
  })

  it('convertInvitationToVcxConnectionCreate', () => {
    const gen = getTestInvitationPayload()
    const invitation = gen.next().value
    if (invitation) {
      expect(
        convertInvitationToVcxConnectionCreate(invitation.payload)
      ).toMatchSnapshot()
    }
  })

  it('convertInvitationToVcxConnectionCreate with version', () => {
    const gen = getTestInvitationPayload()
    const invitation = gen.next().value
    if (invitation) {
      expect(
        convertInvitationToVcxConnectionCreate({
          ...invitation.payload,
          version: '2.0',
        })
      ).toMatchSnapshot()
    }
  })

  it('convertVcxCredentialOfferToCxsClaimOffer', () => {
    expect(
      convertLegacyClaimOfferToAppClaimOffer(vcxClaimOffer)
    ).toMatchSnapshot()
  })
})
