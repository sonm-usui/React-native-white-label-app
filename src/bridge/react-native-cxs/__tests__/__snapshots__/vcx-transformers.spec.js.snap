// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`transformer:VCX convertAgencyConfigToVcxProvision 1`] = `
Object {
  "agency_did": "tjVxL8raUsG5s5ZzGhYV1",
  "agency_url": "https://agency-sandbox.evernym.com",
  "agency_verkey": "VCNhKASjLU5tVWnZpjcyEsoV1QC3adrPVMCCjeKMobn",
  "agent_seed": null,
  "enterprise_seed": null,
  "protocol_type": "4.0",
  "wallet_key": "walletKey",
  "wallet_name": "walletName",
}
`;

exports[`transformer:VCX convertCxsInitToVcxInit 1`] = `
Object {
  "agency_did": "tjVxL8raUsG5s5ZzGhYV1",
  "agency_endpoint": "https://agency-sandbox.evernym.com",
  "agency_verkey": "VCNhKASjLU5tVWnZpjcyEsoV1QC3adrPVMCCjeKMobn",
  "institution_did": "oneTimeAgencyDid",
  "institution_logo_url": "https://try.connect.me/img/CMicon@3x.png",
  "institution_name": "test-name",
  "institution_verkey": "oneTimeAgencyVerKey",
  "protocol_type": "4.0",
  "remote_to_sdk_did": "myOneTimeAgentDid",
  "remote_to_sdk_verkey": "myOneTimeAgentVerificationKey",
  "sdk_to_remote_did": "myOneTimeDid",
  "sdk_to_remote_verkey": "myOneTimeVerificationKey",
  "wallet_key": "walletKey",
  "wallet_name": "walletName",
}
`;

exports[`transformer:VCX convertInvitationToVcxConnectionCreate 1`] = `
Object {
  "invite_details": Object {
    "connReqId": "requestId1",
    "senderAgencyDetail": Object {
      "DID": "senderDID1",
      "endpoint": "endpoint",
      "verKey": "senderVerificationKey1",
    },
    "senderDetail": Object {
      "DID": "senderDID1",
      "agentKeyDlgProof": Object {
        "agentDID": "senderAgentDID1",
        "agentDelegatedKey": "agentDelegatedKey",
        "signature": "signature",
      },
      "logoUrl": "http://testissuer.com/logoUrl.png",
      "name": "sender1",
      "verKey": "senderVerificationKey1",
    },
    "statusCode": "MS-102",
    "statusMsg": "message sent",
    "targetName": "sender1",
  },
  "source_id": "requestId1",
}
`;

exports[`transformer:VCX convertInvitationToVcxConnectionCreate with version 1`] = `
Object {
  "invite_details": Object {
    "connReqId": "requestId1",
    "senderAgencyDetail": Object {
      "DID": "senderDID1",
      "endpoint": "endpoint",
      "verKey": "senderVerificationKey1",
    },
    "senderDetail": Object {
      "DID": "senderDID1",
      "agentKeyDlgProof": Object {
        "agentDID": "senderAgentDID1",
        "agentDelegatedKey": "agentDelegatedKey",
        "signature": "signature",
      },
      "logoUrl": "http://testissuer.com/logoUrl.png",
      "name": "sender1",
      "verKey": "senderVerificationKey1",
    },
    "statusCode": "MS-102",
    "statusMsg": "message sent",
    "targetName": "sender1",
    "version": "2.0",
  },
  "source_id": "requestId1",
}
`;

exports[`transformer:VCX convertVcxCredentialOfferToCxsClaimOffer 1`] = `
Object {
  "claim": Object {
    "Address 1": Array [
      "Address Address Address",
    ],
    "Address 2": Array [
      "Address 2 Address 2 Address 2",
    ],
  },
  "claim_def_id": undefined,
  "claim_name": "Home Address",
  "from_did": "ha66899sadfjZJGINKN0770",
  "issuer_did": "ha66899sadfjZJGINKN0770",
  "msg_type": "CLAIM_OFFER",
  "price": undefined,
  "remoteName": "",
  "schema_seq_no": 36,
  "to_did": "8XFh8yBzrpJQmNyZzgoTqB",
  "version": "1.0.0",
}
`;

exports[`transformer:VCX convertVcxProvisionResultToUserOneTimeInfo 1`] = `
Object {
  "myOneTimeAgentDid": "myOneTimeAgentDid",
  "myOneTimeAgentVerificationKey": "myOneTimeAgentVerificationKey",
  "myOneTimeDid": "myOneTimeDid",
  "myOneTimeVerificationKey": "myOneTimeVerificationKey",
  "oneTimeAgencyDid": "oneTimeAgencyDid",
  "oneTimeAgencyVerificationKey": "oneTimeAgencyVerKey",
}
`;
